tests/provider: WAF data source tests failing on GovCloud

<!--- Please keep this note for the community --->

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->

### Description

These tests fails on GovCloud:

```
TestAccDataSourceAwsWafWebAcl_basic
TestAccDataSourceAwsWafIPSet_basic
TestAccDataSourceAwsWafRateBasedRule_basic
TestAccDataSourceAwsWafRule_basic
TestAccDataSourceAwsWafWebAcl_basic
```

With this error: `no such host`. GovCloud does not support `waf` service.

### New or Affected Resource(s)

<!--- Please list the new or affected resources and data sources. --->

* aws_waf_web_acl
* aws_waf_rule
* aws_waf_rate_based_rule
* aws_waf_ipset

### Terraform Configuration Files

Example config:

```hcl
data "aws_waf_ipset" "ipset" {
  name = aws_waf_ipset.ipset.name
}
```

### Debug Output

```
TestAccDataSourceAwsWafRule_basic: data_source_aws_waf_rule_test.go:17: Step 1/2, expected an error with pattern, no match on: Error running pre-apply refresh: 2020/10/27 05:29:39 [DEBUG] Using modified User-Agent: Terraform/0.12.29 HashiCorp-terraform-exec/0.10.0
        
        Error: error reading WAF Rules: RequestError: send request failed
        caused by: Post "https://waf.us-gov-west-1.amazonaws.com/": dial tcp: lookup waf.us-gov-west-1.amazonaws.com on 192.168.22.2:53: no such host

--- FAIL: TestAccDataSourceAwsWafRule_basic (22.63s)
```

### References

* #15872

--- a/aws/data_source_aws_waf_ipset_test.go
+++ b/aws/data_source_aws_waf_ipset_test.go
@@ -15,7 +15,7 @@ func TestAccDataSourceAwsWafIPSet_basic(t *testing.T) {
 	datasourceName := "data.aws_waf_ipset.ipset"
 
 	resource.ParallelTest(t, resource.TestCase{
-		PreCheck:  func() { testAccPreCheck(t) },
+		PreCheck:  func() { testAccPreCheck(t); testAccPartitionHasServicePreCheck("waf", t) },
 		Providers: testAccProviders,
 		Steps: []resource.TestStep{
 			{

--- a/aws/data_source_aws_waf_rate_based_rule_test.go
+++ b/aws/data_source_aws_waf_rate_based_rule_test.go
@@ -2,10 +2,10 @@ package aws
 
 import (
 	"fmt"
-	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
 	"regexp"
 	"testing"
 
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
 )
 
@@ -15,7 +15,7 @@ func TestAccDataSourceAwsWafRateBasedRule_basic(t *testing.T) {
 	datasourceName := "data.aws_waf_rate_based_rule.wafrule"
 
 	resource.ParallelTest(t, resource.TestCase{
-		PreCheck:  func() { testAccPreCheck(t) },
+		PreCheck:  func() { testAccPreCheck(t); testAccPartitionHasServicePreCheck("waf", t) },
 		Providers: testAccProviders,
 		Steps: []resource.TestStep{
 			{

--- a/aws/data_source_aws_waf_rule_test.go
+++ b/aws/data_source_aws_waf_rule_test.go
@@ -2,10 +2,10 @@ package aws
 
 import (
 	"fmt"
-	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
 	"regexp"
 	"testing"
 
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
 )
 
@@ -15,7 +15,7 @@ func TestAccDataSourceAwsWafRule_basic(t *testing.T) {
 	datasourceName := "data.aws_waf_rule.wafrule"
 
 	resource.ParallelTest(t, resource.TestCase{
-		PreCheck:  func() { testAccPreCheck(t) },
+		PreCheck:  func() { testAccPreCheck(t); testAccPartitionHasServicePreCheck("waf", t) },
 		Providers: testAccProviders,
 		Steps: []resource.TestStep{
 			{

--- a/aws/data_source_aws_waf_web_acl_test.go
+++ b/aws/data_source_aws_waf_web_acl_test.go
@@ -2,10 +2,10 @@ package aws
 
 import (
 	"fmt"
-	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
 	"regexp"
 	"testing"
 
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
 )
 
@@ -15,7 +15,7 @@ func TestAccDataSourceAwsWafWebAcl_basic(t *testing.T) {
 	datasourceName := "data.aws_waf_web_acl.web_acl"
 
 	resource.ParallelTest(t, resource.TestCase{
-		PreCheck:  func() { testAccPreCheck(t) },
+		PreCheck:  func() { testAccPreCheck(t); testAccPartitionHasServicePreCheck("waf", t) },
 		Providers: testAccProviders,
 		Steps: []resource.TestStep{
 			{

